presto-hive/src/main/java/com/facebook/presto/hive/util/StatisticsUtils.java

```diff
@@ -0,0 +1,78 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.util;
+
+import com.facebook.presto.hive.HiveBasicStatistics;
+import com.facebook.presto.hive.metastore.Partition;
+import com.facebook.presto.hive.metastore.Table;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Collections.unmodifiableMap;
+
+public class StatisticsUtils
+{
+    private StatisticsUtils() {}
+
+    /**
+     * This method is intended to be used to increment statistics for an existing table
+     * <p>
+     * If some of the existing table statistics are missing this method does not overwrite them
+     * <p>
+     * This method throws an exception when some of the updated statistics are not present
+     * We assume that whenever new data is inserted, all statistics for that data must be collected
+     * <p>
+     * This comment applies for both, incrementStatisticsCounters and incrementStatisticsCounters methods
+     */
+    public static Table incrementStatisticsCounters(Table table, HiveBasicStatistics statistics)
+    {
+        //
+        checkAllStatisticsArePresent(statistics);
+        Map<String, String> parameters = table.getParameters();
+        Map<String, String> updatedParameters = incrementStatisticsCounters(parameters, statistics);
+        return Table.builder(table)
+                .setParameters(updatedParameters)
+                .build();
+    }
+
+    public static Partition incrementStatisticsCounters(Partition partition, HiveBasicStatistics statistics)
+    {
+        // whenever new data is inserted all statistics for that data must be collected
+        checkAllStatisticsArePresent(statistics);
+        Map<String, String> parameters = partition.getParameters();
+        Map<String, String> updatedParameters = incrementStatisticsCounters(parameters, statistics);
+        return Partition.builder(partition)
+                .setParameters(updatedParameters)
+                .build();
+    }
+
+    private static Map<String, String> incrementStatisticsCounters(Map<String, String> parameters, HiveBasicStatistics statistics)
+    {
+        HiveBasicStatistics currentStatistics = HiveBasicStatistics.createFromPartitionParameters(parameters);
+        HiveBasicStatistics updatedStatistics = currentStatistics.add(statistics);
+        Map<String, String> updatedParameters = new HashMap<>(parameters);
+        updatedParameters.putAll(updatedStatistics.toPartitionParameters());
+        return unmodifiableMap(updatedParameters);
+    }
+
+    private static void checkAllStatisticsArePresent(HiveBasicStatistics statistics)
```

Comment:
maybe you could move this method to `HiveBasicStatistics`

Reply:
Why? Do we usually do this? We rather have `check(some_obejct_to_check)` than `some_obejct_to_check.check()`. Let me know if i didn't get your point right.

Reply:
Oh, got you know =) I'm slow. Yeah, that makes sense. Will do.

Reply:
I just checked the code. I don't want to do `checkArgument(statistics.isAllStatisticsPresent)` because i want to have granular error message (like what statistic is missing).